Sources/PenguinStructures/Graphs/AdjacencyList.swift

```diff
@@ -85,12 +88,26 @@ extension AdjacencyList: MutableGraph {
 		return EdgeId(source: source, destination: destination, offset: offset)
 	}
 
-	public mutating func removeEdge(from u: VertexId, to v: VertexId) {
+    /// Removes the edge (u, v) from the graph.
+    ///
+    /// If there are parallel edges, it removes all edges.
+    ///
+    /// - Precondition: `u` and `v` are valid `VertexId`s from `self`.
+    /// - Throws: `GraphErrors.edgeNotFound` if no edges are found.
+    /// - Complexity: O(E) or faster.
+	public mutating func removeEdge(from u: VertexId, to v: VertexId) throws {
 		assertValid(u, name: "u")
 		assertValid(v, name: "v")
+		let previousEdgeCount = edgesArray[u.index].count
 		edgesArray[u.index].removeAll { $0 == v.id }
+		if previousEdgeCount == edgesArray[u.index].count {
+			throw GraphErrors.edgeNotFound
+		}
 	}
 
+    /// Removes the edge `edge` from the graph.
+    ///
+    /// - Precondition: `edge` is a valid `EdgeId` from `self`.
 	public mutating func remove(edge: EdgeId) {
```

Comment:
Given the mismatch here it might be a good time to switch the whole project to 2-space indents/ swift-format it.

Reply:
Yeah, I want to just auto-format everything. Just haven't gotten to it yet. I'll try and do that before tomorrow.